### PR TITLE
Modify entity, Add register and retrieve APIs of Store product

### PR DESCRIPTION
**Pull request by ZinnaChoi**
Hi @JoshRyu 

I've updated the JPA entities to reflect changes in the ERD. 
Additionally, I've implemented APIs for registering and retrieving store products. 

Thank you! 🙂🙂

### CODE REVIEW

File: backend/src/main/java/com/madeg/logistics/service/StoreProductService.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    Integer income = storeProductInput.getIncomeCnt() == null
+      ? 0
+      : storeProductInput.getIncomeCnt();
+
+    StoreProduct storeProduct = StoreProduct
+      .builder()
+      .store(store)
+      .product(product)
+      .storePrice(
+        storeProductInput.getStorePrice() == null
```
Comment (JoshRyu) [inline backend/src/main/java/com/madeg/logistics/service/StoreProductService.java:83]: Looks great! it deals with various exceptions that possibly occur during the store product registration process.
There is one thing I concern about, can storeProductInput.getStorePrice() have negative value?

**Review by JoshRyu — COMMENTED** [review]
Overall well-documented and well-featured PR. 
Thanks for your contribution. there is one comment, so please check it out!

File: backend/src/main/java/com/madeg/logistics/service/StoreProductService.java (hunk 1 above)
Comment (ZinnaChoi) [reply backend/src/main/java/com/madeg/logistics/service/StoreProductService.java:83]: I've added the @DecimalMin annotation to the store_price field in StoreProductInput.java. This ensures that negative values are not allowed. For instance, if you make a request as shown below, you'll receive an error message indicating that negative values are not permitted.

image

Additionally, upon reviewing other entities' input and patch files, I noticed that the product's price didn't have a minimum value constraint. So, I've implemented this constraint there as well.

Thank you!

**Review by JoshRyu — APPROVED** [review]
Looks good thanks!

**Pull request merged**